tests/integrations/fivetran/test_fivetran.py

```diff
@@ -0,0 +1,655 @@
+from unittest import mock
+
+import pytest
+import requests_mock
+from airflow.exceptions import AirflowException
+
+from universal_transfer_operator.datasets.file.base import File
+from universal_transfer_operator.datasets.table import Table
+from universal_transfer_operator.integrations.fivetran.connector.aws.s3 import S3Connector
+from universal_transfer_operator.integrations.fivetran.connector.base import FivetranConnector
+from universal_transfer_operator.integrations.fivetran.destination.base import FivetranDestination
+from universal_transfer_operator.integrations.fivetran.fivetran import (
+    FivetranIntegration,
+    FiveTranOptions,
+    Group,
+)
+
+
+class TestFivetranIntegration:
+    @mock.patch("universal_transfer_operator.integrations.fivetran.fivetran.FivetranIntegration.hook")
+    def test_transfer_using_connector_id(self, mock_hook):
+        """Test the transfer using connector id."""
+        mock_hook.prep_connector.return_value = True
+        mock_hook.start_fivetran_sync.return_value = True
+
+        fivetran_integrations = FivetranIntegration()
+        assert fivetran_integrations.transfer_using_connector_id(connector_id="dummy") is True
+
+    @mock.patch("universal_transfer_operator.integrations.fivetran.fivetran.FivetranIntegration.create_group")
+    @mock.patch(
+        "universal_transfer_operator.integrations.fivetran.fivetran.FivetranIntegration.check_group_details"
+    )
+    def test_create_group_if_needed(self, mock_group, mock_check_group_details):
+        """Test to create group using group_id"""
```

Comment:
I think we can convert these integration tests to unit test cases. We can make sure the Filetran API is called with appropriate parameters. In both cases
1. When we need to create a group
2. When we just check to find out that it already exists.
WDYT?